plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/wizard/dependencies/SetupOptionsPage.java

```diff
@@ -0,0 +1,196 @@
+		TableItem defaultItem = new TableItem(table, SWT.NONE);
+		defaultItem.setText(0, "default");
+		defaultItem.setText(1, "default settings");
+
+		data.forEach((name, path) -> {
+			TableItem item = new TableItem(table, SWT.NONE);
+			item.setText(0, name);
+			item.setText(1, path);
+		});
+
+		table.getColumn(0).pack();
+		table.getColumn(1).pack();
+		table.pack(true);
+		return table;
+	}
+
+	/**
+	 * Adds new controls to the UI, that allow user to select dynamically calculated configurations for pending
+	 * operations.
+	 */
+	private void updateConfigs(Map<String, String> npmrcs) {
+
+		configsContainer.setVisible(false);
+		for (Control kid : configsContainer.getChildren()) {
+			kid.dispose();
+		}
+		configsContainer.pack(true);
+
+		tNPMRC = createTable(configsContainer, npmrcs, ".npmrc configurations",
+				"Please select configuration for '.npmrc'.");
+
+		container.layout();
+		configsContainer.setVisible(true);
+	}
+
+	/** On page entry (re)discover config files. */
+	@Override
+	public void setVisible(boolean visible) {
+		setPageComplete(visible);
+		if (visible)
+			discoverConfigFilesAndUpdate();
+		super.setVisible(visible);
+	}
+
+	/** Saves user choices into provided object */
+	public void saveOptions(InstallOptions options) {
+		if (tNPMRC != null)
+			options.npmrc = getTableItem(tNPMRC);
+		if (cleanNpmCache != null)
+			options.clearNpmCache = cleanNpmCache.getSelection();
+	}
+
+	/** Scan file system for the relevant {@code .npmrc} files. */
+	private void discoverConfigFilesAndUpdate() {
+		RunnableSettingsFilesLocator runnableSettingsFilesLocator = new RunnableSettingsFilesLocator();
+		try {
+			getContainer().run(true, true, runnableSettingsFilesLocator);
+		} catch (InvocationTargetException | InterruptedException e) {
+			LOGGER.error("Error during configuration files discovery.", e);
+			UIUtils.getDisplay().asyncExec(() -> MessageDialog.openError(
+					UIUtils.getShell(),
+					"Error during configuration files discovery.",
+					"Error scanning files system for the configuration files..\n"
+							+ "Please check your Error Log view for the detailed log about the failure."));
+		}
+
+		Collection<File> fNPMRCs = runnableSettingsFilesLocator.getCollectedConfigFiles();
+
+		Map<String, String> npmrcs = new HashMap<>();
+		fNPMRCs.forEach(f -> npmrcs.put(f.getName(), f.getAbsolutePath()));
+		updateConfigs(npmrcs);
+	}
+
+	/** Expects one row selected in the table and returns value of second row. */
+	private String getTableItem(Table table) {
+		String result = "";
+		if (table == null)
+			return result;
+
+		TableItem[] items = table.getSelection();
+
+		if (items.length >= 2) {
+			String text = Strings.nullToEmpty(table.getToolTipText());
+			if (!text.isEmpty())
+				text = " :: " + text;
+			throw new RuntimeException("Multiple selections are not supported" + text);
+		}
+
+		if (items.length == 1)
+			if (items[0] != null)
+				// first is the display name, second is data
```

Comment:
omitting braces... well, but two times in a row? and without an else block 😕

Reply:
I don't like `else` parts, prefer pattern matching or `switch` cases if the are at least like in Xtend.
I don't like braces, semicolons or other things required to reason about code when you print it in black and white.
I want language that is whitespace sensitive *and* editor that does expression+scope based colouring.


Anyway, do you feel like adding `{{}}` is necessary? I can add it.

Reply:
I meant the combination of nested if-statements _and_ having no braces makes it strange. But I see your point with the pattern above. Do as you prefer.

Reply:
or `if (items.length == 1 && items[0] != null)`

Reply:
then let's do it like @kduske-n4